app/Fields/Currency.php

```diff
@@ -91,6 +91,27 @@ public static function getCurrencyIdByName($currencyName)
 		return $currencyId;
 	}
 
+	/**
+	 * Get currency by code.
+	 *
+	 * @param string $currencyCode
+	 *
+	 * @return int|null
+	 */
+	public static function getCurrencyIdByCode(string $currencyCode): ?int
+	{
+		$currencyId = null;
+		$row = (new \App\Db\Query())
+			->select(['id'])
+			->from('vtiger_currency_info')
+			->where(['currency_code' => \strtoupper($currencyCode), 'deleted' => 0])
+			->scalar();
+		if ($row) {
+			$currencyId = $row;
+		}
+		return $currencyId;
```

Comment:
1. Change name to getIdByCode()
2. Add arg $active = true
2. Replace the function's contents
    $currency = array_column(\App\Fields\Currency::getAll($active), 'id', 'currency_code');
    return $currency[\strtoupper($currencyCode)]??null;
    or add foreach